src/common/translate_item.js

```diff
@@ -105,8 +105,52 @@ Zotero.Translate.ItemSaver.prototype = {
 			payload.cookie = document.cookie;
 		}
 		payload.proxy = this._proxy && this._proxy.toJSON();
+		
+		// Add page data for snapshot
+		// TODO: Check for skipSnapshot
+		let singleFile;
+		for (let i = 0; i < items.length; i++) {
+			if (items[i].itemType !== 'webpage') {
+				continue;
+			}
+
+			for (let j = 0; j < items[i].attachments.length; j++) {
+				if (items[i].attachments[j].mimeType !== 'text/html') {
```

Comment:
A snapshot is indicated by `mimeType: 'text/html'` and `snapshot !== false`. (`snapshot: false` is a linked-URL attachment. If `snapshot` is missing, that implies `true`.)